### PR TITLE
Add 'last update was a complaint' filter for cases.

### PR DESCRIPTION
**Pull request by neprune**
Issue: https://github.com/mysociety/societyworks/issues/3093

Filter will ignore existing cases where the last update is a complaint.
I could fix this by writing a migration to back-populate the `last_update_type` based on the created times for a case's actions if we want to.

### CODE REVIEW

**Comment by neprune** [comment]
Filter looks like:
Screenshot 2022-09-26 at 14 12 42

**Review by dracos — APPROVED** [review]
This seems like a nice way of doing it 👍

**Comment by dracos** [comment]
Probably is worth writing the migration, yeah, would prevent possible confusion I assume.

**Comment by neprune** [comment]
Added the migration to populate the last update type for existing cases. Slow approach but don't think it should be problematic.

File: cases/migrations/0029_back_populate_case_last_update_type.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        # just ignore it.
+        return
+
+    case.save()
+
+
+def forwards_func(apps, schema_editor):
+    Case = apps.get_model("cases", "Case")
+
+    for case in Case.objects.all():
```
Comment (dracos) [inline cases/migrations/0029_back_populate_case_last_update_type.py:39]: I think you can use `Case.objects.annotate(Max("complaints__created"), Max("actions__created"))` here which would then give you all the info you want in one query (each case would have `actions__created__max` and `complaints__created__max` attributes).

**Review by dracos — CHANGES_REQUESTED** [review]
I think you can optimise it all into one db call

File: cases/migrations/0029_back_populate_case_last_update_type.py (hunk 1 above)
Comment (neprune) [reply cases/migrations/0029_back_populate_case_last_update_type.py:39]: Nice!

**Comment by neprune** [comment]
History is a bit of a mess; will merge as a single squashed commit.

File: cases/migrations/0031_populate_case_last_update_type.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+                            last_action_time__gt=F("last_complaint_time"),
+                        ),
+                        then=Value(ACTION_TYPE),
+                    ),
+                    default=Value(NONE_TYPE),
+                )
+            )
+            .values("_last_update_type")[:1]
+        )
+    )
```
Comment (neprune) [inline cases/migrations/0031_populate_case_last_update_type.py:49]: Populate migration is now a single query.

**Review by dracos — APPROVED** [review]
I was only thinking of adding the annotation and still having a loop, but this is nice ;)

**Pull request merged**